Add Files, Clipboard, and first Advanced Apps Patterns (#7645)
GoogleChrome/web.dev: src/site/content/en/patterns/advanced-apps/contacts/index.md

---
layout: pattern
title: How to access contacts from the address book
date: 2022-10-06
authors:
  - thomassteiner
description: >
  Learn how to get contacts from the address book.
height: 800
---

Sometimes you want to let users of your app select one of their contacts to message via an email or
chat application, or help them discover which of their contacts have already joined a social
platform.

## The modern way

### Using the Contact Picker API

To get contacts from the address book, you need to use the
[Contact Picker API](https://developer.mozilla.org/docs/Web/API/Contact_Picker_API), which allows
users to select entries from their contact list and share limited details of the selected entries
with your app. Several properties like `name`, `email`, `tel`, `address`, and `icon` are available.
To find out about the concretely supported properties, call `navigator.contacts.getProperties()`. To
allow the user to select multiple contacts, pass `{multiple: true}` as the second parameter of
`navigator.contacts.select()`.

{% BrowserCompat 'api.AudioContext.outputLatency' %}

> The Contact Picker API is available in the Android version of Chrome starting in version 80 and up.

Reviewer: ```suggestion
The Contact Picker API is available in the Android version of Chrome from version 80.
```
